src/algebra/category/Module/change_of_rings.lean

```diff
@@ -20,6 +20,10 @@ import ring_theory.tensor_product
   module structure is defined by `s • (s' ⊗ m) := (s * s') ⊗ m` and `R`-linear map `l : M ⟶ M'`
   is sent to `S`-linear map `s ⊗ m ↦ s ⊗ l m : S ⨂ M ⟶ S ⨂ M'`.
 
+* `category_theory.Module.coextend_scalars`: given rings `R, S` and a ring homomorphism `R ⟶ S`
+  then `coextend_scalars : Module R ⥤ Module S` is defined by `M ↦ (S →ₗ[R] M)` where `S` is seen as
+  `R-module` by restriction of scalars and `l ↦ l ∘ _`.
```

Comment:
Please mention that it is a right adjoint to restriction of scalars. Can you also prove that?

Reply:
done in [5ec3cdd](https://github.com/leanprover-community/mathlib/pull/15958/commits/5ec3cdd508d26e324553e19f48458ea5ee24d7fe)